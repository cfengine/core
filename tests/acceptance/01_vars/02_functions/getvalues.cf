#######################################################
#
# Test getvalues()
#
#######################################################

body common control
{
      inputs => { "../../default.cf.sub" };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}

#######################################################

bundle agent test
{
  vars:
      "user[name]"              string => "zamboni";
      "user[fullname][first]"   string => "Diego";
      "user[fullname][last]"    string => "Zamboni";
      "user[dirs]"              slist => { "/home/zamboni",
                                           "/tmp/zamboni",
                                           "/export/home/zamboni" };

      "values"     slist => getvalues("user");
      "uservalues" slist => getvalues("user[fullname]");
      "inline_values_map" slist => getvalues('{ "foo": 1, "bar": 2 }');
      "inline_values_array" slist => getvalues('[ "foo", 1, "bar", 2 ]');
}

#######################################################

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
