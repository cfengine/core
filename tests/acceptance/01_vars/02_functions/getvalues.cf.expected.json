{
  "inline_values_array": [
    "foo",
    "1",
    "bar",
    "2"
  ],
  "inline_values_map": [
    "1",
    "2"
  ],
  "user[dirs]": [
    "/home/zamboni",
    "/tmp/zamboni",
    "/export/home/zamboni"
  ],
  "user[fullname][first]": "Diego",
  "user[fullname][last]": "Zamboni",
  "user[name]": "zamboni",
  "uservalues": [
    "Diego",
    "Zamboni"
  ],
  "values": [
    "Diego",
    "Zamboni",
    "/home/zamboni",
    "/tmp/zamboni",
    "/export/home/zamboni",
    "zamboni"
  ]
}
